packages/operator/api/v1alpha1/batchinferencejob_types.go

```diff
@@ -0,0 +1,125 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BatchJobState defines current state
+type BatchJobState string
+
+const (
+	BatchScheduling BatchJobState = "scheduling"
+	BatchRunning    BatchJobState = "running"
+	BatchSucceeded  BatchJobState = "succeeded"
+	BatchFailed     BatchJobState = "failed"
+	BatchUnknown    BatchJobState = "unknown"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// BatchInferenceJobSpec defines the desired state of BatchInferenceJob
+type BatchInferenceJobSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	// Docker image
+	Image string `json:"image"`
+	// Entrypoint array. Not executed within a shell.
+	// The docker image's ENTRYPOINT is used if this is not provided.
+	// Variable references $(VAR_NAME) are expanded using the container's environment. If a variable
+	// cannot be resolved, the reference in the input string will be unchanged. The $(VAR_NAME) syntax
+	// can be escaped with a double $$, ie: $$(VAR_NAME). Escaped references will never be expanded,
+	// regardless of whether the variable exists or not.
+	// Cannot be updated.
+	// More info: https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell
+	Command []string `json:"command"`
+	// Arguments to the entrypoint.
+	// The docker image's CMD is used if this is not provided.
+	// Variable references $(VAR_NAME) are expanded using the container's environment. If a variable
+	// cannot be resolved, the reference in the input string will be unchanged. The $(VAR_NAME) syntax
+	// can be escaped with a double $$, ie: $$(VAR_NAME). Escaped references will never be expanded,
+	// regardless of whether the variable exists or not.
+	// Cannot be updated.
+	// More info: https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#running-a-command-in-a-shell
+	Args []string `json:"args"`
+	// InputConnection is name of connection to object storage bucket where input data are expected
+	InputConnection string `json:"inputConnection"`
+	// InputPath is a source directory for BatchInferenceJob input data
+	// relative to bucket root of InputConnection
+	InputPath string `json:"inputPath"`
```

Comment:
_Minor_
Intuitively, it seems attractive to create a small special structure like:
```
type ObjectLocation struct {
  ConnectionID string
  Path string
}
```
and use it for `Input`, `Output` and `Model`.

Reply:
Let's leave it as is